torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4973,9 +4965,10 @@ def gen_inputs():
                     # Either A or B needs to have a gradient
                     if not grad_A and not grad_B:
                         continue
-                    A.requires_grad_(grad_A)
-                    B.requires_grad_(grad_B)
-                    yield SampleInput(A, args=(B,), kwargs=kwargs)
+                    yield SampleInput(
+                        A.clone().requires_grad_(grad_A),
```

Comment:
Why is the clone needed here?

Reply:
@mruberry GitHub cuts this off in the diff, but it's inside a loop
```
for grad_A, grad_B in product((True, False), repeat=2):
  # Either A or B needs to have a gradient
  if not grad_A and not grad_B:
    continue
  A.requires_grad_(grad_A)
  B.requires_grad_(grad_B)
  yield SampleInput(A, args=(B,), kwargs=kwargs)
```

So previously the same tensor was returned in multiple `SampleInputs` simultaneously which causes this assert to fail
https://github.com/pytorch/pytorch/blob/fc8404b5bc7e9721aa93021cdc27de818df64d8d/test/test_ops.py#L251

Reply:
This clone is required to make inputs independent yet with the same content.